src/dune/action_exec.ml

```diff
@@ -3,6 +3,9 @@ open Import
 open Fiber.O
 module DAP = Dune_action_plugin.Private.Protocol
 
+(* This feels like the wrong place for this, but I'm not sure of a better one. *)
+let terminal_lock = Fiber.Mutex.create ()
```

Comment:
Looks about right to me. The terminal is a global resource, so it seems reasonable to use a global lock.

Reply:
However, yes this might not be the right module to put it in.